Regression in v1.12.0 release

<!--
  ⚡️ katchow! We 💛 issues.

  Please - do not - remove this template.
  Please - do not - skip or remove parts of this template.
  Or your issue may be closed.

  👉🏽 Need help or tech support? Please don't open an issue!
  Head to https://gitter.im/rollup/rollup or https://stackoverflow.com/questions/tagged/rollupjs

  ❤️ Rollup? Please consider supporting our collective:
  👉 https://opencollective.com/rollup/donate
-->

- Rollup Version: `v1.12.0`
- Operating System (or Browser): Ubuntu 19.04; `Linux max 5.0.0-15-generic #16-Ubuntu SMP Mon May 6 17:41:33 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
- Node Version: tested with `v10.15.3`, `v11.15.0`, and `v12.2.0`

### How Do We Reproduce?

<!--
  Issues without minimal reproductions will be closed! Please provide one by:
  1. Using the REPL at https://rollupjs.org/repl, or
  2. Work to isolate the problem and provide the exact steps in this issue, or
  3. Provide a minimal repository link (Read https://git.io/fNzHA for instructions).
     These may take more time to triage than the other options.
-->

1. For an existing project which breaks see <https://github.com/MaxMilton/new-tab>.
1. Clone a copy of master locally and install dependencies.
1. Run either `yarn run build` or `yarn run dev`, both break in different ways 😬

Both `npm` and `yarn` produce the same result.

I'll to create a more minimal repo in a moment and try dig a little deeper into what might be going wrong.

### Expected Behavior

Build runs to completion as in rollup `v1.11.3`.

### Actual Behavior

_Note: This is from the project mentioned above which uses a bunch of rollup plugins so the following might not be very concise. Will post back once I can repro with a minimal example._

<details>
<summary>`yarn run dev` (expand me):</summary>

```sh
> new-tab@0.14.0 dev /media/max/xps-linux/home/max/Projects/new-tab
> rollup --config --watch

[!] ReferenceError: Cannot access 'processConfigsErr' before initialization
ReferenceError: Cannot access 'processConfigsErr' before initialization
    at /media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/bin/rollup:1457:35
    at Array.map (<anonymous>)
    at processConfigs (/media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/bin/rollup:1441:24)
    at watch (/media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/bin/rollup:1433:28)
    at execute (/media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/bin/rollup:1662:9)
    at /media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/bin/rollup:1653:30

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! new-tab@0.14.0 dev: `rollup --config --watch`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the new-tab@0.14.0 dev script.
```
</details>

<details>
<summary>`yarn run build`  (expand me):</summary>

```sh
yarn run v1.16.0
$ rm -rf dist && mkdir dist && cp icon* dist
$ NODE_ENV=production rollup --config

src/app.ts → dist/n.js...
[!] Error: Unexpected token (Note that you need plugins to import files that are not JavaScript)
src/App.css (1:0)
1: :root{--s:0 2px 4px rgba(0,0,17,0.15);--b:#fafafa;--t:#383a42;--c:#e5e5e6;--c0:#fff;--c1:#dbdbdc;--c2:#eef0f2}.d{--b:#23252d;--t:#9da5b4;--c:#21252b;--c0:#292d35;--c1:#383e4a;--c2:#292d35}body{margin:80px 0;font-size:18px;line-height:41px;text-rendering:optimizeSpeed;font-variant-ligatures:none;color:var(--t);background-color:var(--b)}.muted,h2{color:var(--c1)}a{text-decoration:none}a:focus,a:hover{text-decoration:underline}
   ^
2: /*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjpudWxsLCJzb3VyY2VzIjpbIi9tZWRpYS9tYXgveHBzLWxpbnV4L2hvbWUvbWF4L1Byb2plY3RzL25ldy10YWIvc3JjL0FwcC5zdmVsdGUiXSwic291cmNlc0NvbnRlbnQiOlsiPHNjcmlwdD5cbiAgaW1wb3J0IHsgaGFuZGxlTGlua0NsaWNrIH0gZnJvbSAnLi9jb21tb24nO1xuICBpbXBvcnQgQm9va21hcmtHcm91cCBmcm9tICcuL2NvbXBvbmVudHMvQm9va21hcmtHcm91cC5zdmVsdGUnO1xuICBpbXBvcnQgTWVudSBmcm9tICcuL2NvbXBvbmVudHMvTWVudS5zdmVsdGUnO1xuICBpbXBvcnQgU2VhcmNoIGZyb20gJy4vY29tcG9uZW50cy9TZWFyY2guc3ZlbHRlJztcblxuICB3aW5kb3cuYWRkRXZlbnRMaXN0ZW5lcignY2xpY2snLCBoYW5kbGVMaW5rQ2xpY2ssIHRydWUpO1xuPC9zY3JpcHQ+PHN0eWxlIHR5cGU9XCJ0ZXh0L3Bvc3Rjc3NcIj46Z2xvYmFsKDpyb290KXstLXM6MCAycHggNHB4IHJnYmEoMCwwLDE3LDAuMTUpOy0tYjojZmFmYWZhOy0tdDojMzgzYTQyOy0tYzojZTVlNWU2Oy0tYzA6I2ZmZjstLWMxOiNkYmRiZGM7LS1jMjojZWVmMGYyfTpnbG9iYWwoLmQpey0tYjojMjMyNTJkOy0tdDojOWRhNWI0Oy0tYzojMjEyNTJiOy0tYzA6IzI5MmQzNTstLWMxOiMzODNlNGE7LS1jMjojMjkyZDM1fTpnbG9iYWwoYm9keSl7bWFyZ2luOjgwcHggMDtmb250LXNpemU6MThweDtsaW5lLWhlaWdodDo0MXB4O3RleHQtcmVuZGVyaW5nOm9wdGltaXplU3BlZWQ7Zm9udC12YXJpYW50LWxpZ2F0dXJlczpub25lO2NvbG9yOnZhcigtLXQpO2JhY2tncm91bmQtY29sb3I6dmFyKC0tYil9Omdsb2JhbCgubXV0ZWQpLDpnbG9iYWwoaDIpe2NvbG9yOnZhcigtLWMxKX06Z2xvYmFsKGEpe3RleHQtZGVjb3JhdGlvbjpub25lfTpnbG9iYWwoYSk6Zm9jdXMsOmdsb2JhbChhKTpob3Zlcnt0ZXh0LWRlY29yYXRpb246dW5kZXJsaW5lfTwvc3R5bGU+PEJvb2ttYXJrR3JvdXAvPjxTZWFyY2gvPjxNZW51Lz4iXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBTzRDLEtBQUssQUFBQyxDQUFDLElBQUksMkJBQTJCLENBQUMsSUFBSSxPQUFPLENBQUMsSUFBSSxPQUFPLENBQUMsSUFBSSxPQUFPLENBQUMsS0FBSyxJQUFJLENBQUMsS0FBSyxPQUFPLENBQUMsS0FBSyxPQUFPLENBQUMsQUFBUSxFQUFFLEFBQUMsQ0FBQyxJQUFJLE9BQU8sQ0FBQyxJQUFJLE9BQU8sQ0FBQyxJQUFJLE9BQU8sQ0FBQyxLQUFLLE9BQU8sQ0FBQyxLQUFLLE9BQU8sQ0FBQyxLQUFLLE9BQU8sQ0FBQyxBQUFRLElBQUksQUFBQyxDQUFDLE9BQU8sSUFBSSxDQUFDLENBQUMsQ0FBQyxVQUFVLElBQUksQ0FBQyxZQUFZLElBQUksQ0FBQyxlQUFlLGFBQWEsQ0FBQyx1QkFBdUIsSUFBSSxDQUFDLE1BQU0sSUFBSSxHQUFHLENBQUMsQ0FBQyxpQkFBaUIsSUFBSSxHQUFHLENBQUMsQ0FBQyxBQUFRLE1BQU0sQUFBQyxDQUFDLEFBQVEsRUFBRSxBQUFDLENBQUMsTUFBTSxJQUFJLElBQUksQ0FBQyxDQUFDLEFBQVEsQ0FBQyxBQUFDLENBQUMsZ0JBQWdCLElBQUksQ0FBQyxBQUFRLENBQUMsQUFBQyxNQUFNLENBQUMsQUFBUSxDQUFDLEFBQUMsTUFBTSxDQUFDLGdCQUFnQixTQUFTLENBQUMifQ== */
Error: Unexpected token (Note that you need plugins to import files that are not JavaScript)
    at error (/media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/dist/rollup.js:3410:30)
    at Module.error (/media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/dist/rollup.js:12900:9)
    at tryParse (/media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/dist/rollup.js:12817:16)
    at Module.setSource (/media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/dist/rollup.js:13089:35)
    at /media/max/xps-linux/home/max/Projects/new-tab/node_modules/rollup/dist/rollup.js:15906:20
    at async Promise.all (index 5)
    at async Promise.all (index 0)
    at async Promise.all (index 0)
    at async Promise.all (index 0)
    at async Promise.all (index 0)

error Command failed with exit code 1.
```
</details>

<!--
  Most issues can be expressed or demonstrated through the REPL or a repository.
  However, the situation may arise where some small code snippets also need to
  be provided. In that situation, please add your code below using
  Fenced Code Blocks (https://help.github.com/articles/creating-and-highlighting-code-blocks/)
-->

--- a/src/utils/transform.ts
+++ b/src/utils/transform.ts
@@ -76,21 +76,21 @@ export default function transform(
 			}
 		}
 
-		if (!result) return code;
-
 		if (typeof result === 'string') {
 			result = {
 				ast: undefined,
 				code: result,
 				map: undefined
 			};
-		} else {
+		} else if (result && typeof result === 'object') {
 			if (typeof result.map === 'string') {
 				result.map = JSON.parse(result.map);
 			}
 			if (typeof result.moduleSideEffects === 'boolean') {
 				moduleSideEffects = result.moduleSideEffects;
 			}
+		} else {
+			return code;
 		}
 
 		if (result.map && typeof (result.map as ExistingRawSourceMap).mappings === 'string') {

--- a/test/function/samples/transform-empty-string/_config.js
+++ b/test/function/samples/transform-empty-string/_config.js
@@ -1,0 +1,21 @@
+const assert = require('assert');
+
+const sideEffects = [];
+
+module.exports = {
+	description: 'allows transformers to transform code to an empty string',
+	context: { sideEffects },
+	exports() {
+		assert.deepStrictEqual(sideEffects, ['this happens']);
+	},
+	options: {
+		plugins: {
+			name: 'test-plugin',
+			transform(code, id) {
+				if (id.endsWith('transformed.js')) {
+					return '';
+				}
+			}
+		}
+	}
+};

--- a/test/function/samples/transform-empty-string/main.js
+++ b/test/function/samples/transform-empty-string/main.js
@@ -1,0 +1,3 @@
+import './transformed.js';
+
+sideEffects.push('this happens');

--- a/test/function/samples/transform-empty-string/transformed.js
+++ b/test/function/samples/transform-empty-string/transformed.js
@@ -1,0 +1,1 @@
+sideEffects.push('should not happen');

